core/src/main/scala/akka/kafka/internal/SingleSourceLogic.scala

```diff
@@ -67,6 +67,7 @@ private[kafka] abstract class SingleSourceLogic[K, V, Msg](
   }
 
   val partitionAssignedCB = getAsyncCallback[Iterable[TopicPartition]] { newTps =>
+    requested = false
```

Comment:
I don't this is the correct thing to do. As far as I can see the interaction between the `SingleSourceLogic` and the `KafkaConsumerActor` is designed to have at most one outstanding `RequestMessages` in flight. This can result in that another `RequestMessages` is sent to the actor and I think that can result in weird things. I think we must find another solution.

A reproducer is a good start.